src/internal/_Maybe.js

```diff
@@ -0,0 +1,66 @@
+var equals = require('../equals');
+var identity = require('../identity');
+var toString = require('../toString');
+
+
+module.exports = (function() {
+
+  function Maybe() {}
+  Maybe.prototype.of = function _maybeof(x) { return new Just(x); };
+  Maybe.prototype['@@type'] = 'ramda/Maybe';
+  Maybe.prototype.equals = function _maybeequals(that) {
+    return that != null &&
+      that['@@type'] === this['@@type'] &&
+      equals(this.value, that.value);
+  };
+
+  function Just(x) { this.value = x; }
+  Just.prototype = new Maybe();
+  Just.prototype.map = function _justmap(f) {
+    return new Just(f(this.value));
+  };
+  Just.prototype.ap = function _justap(m) {
+    return m.map(this.value);
+  };
+  Just.prototype.chain = function _justchain(f) {
+    return f(this.value);
+  };
+  Just.prototype.reduce = function _justreduce(f, x) {
+    return f(x, this.value);
+  };
+  Just.prototype.getOrElse = function _getOrElse() {
+    return this.value;
+  };
+  Just.prototype.toString = function() {
+    return 'Just(' + toString(this.value) + ')';
+  };
+  Just.prototype.isNothing = false;
+  Just.prototype.isJust = true;
+
+  function returnThis() { return this; }
+
+  function Nothing() {}
```

Comment:
funny that is the approach ramda-fantasy took; that there is a single `Nothing`. But once it's released in the wild it can get mutated by naughty developers. I feel better guaranteeing that `Nothing()` will return a well-behaved `Nothing` at the risk of many `Nothing`s